examples/all-clusters-app/ameba/main/include/ManualOperationalStateCommand.h

```diff
@@ -0,0 +1,213 @@
+}
+
+CHIP_ERROR ManualOperationalStateSetErrorCommandHandler(int argc, char ** argv)
+{
+    if (argc != 1)
+    {
+        return ManualOperationalStateCommandHelpHandler(argc, argv);
+    }
+
+    GenericOperationalError err(to_underlying(ErrorStateEnum::kNoError));
+    uint32_t error = atoi(argv[0]);
+
+    switch (error)
+    {
+    case to_underlying(OperationalState::ErrorStateEnum::kNoError):
+        err.errorStateID = to_underlying(OperationalState::ErrorStateEnum::kNoError);
+        break;
+    case to_underlying(OperationalState::ErrorStateEnum::kUnableToStartOrResume):
+        err.errorStateID = to_underlying(OperationalState::ErrorStateEnum::kUnableToStartOrResume);
+        break;
+    case to_underlying(OperationalState::ErrorStateEnum::kUnableToCompleteOperation):
+        err.errorStateID = to_underlying(OperationalState::ErrorStateEnum::kUnableToCompleteOperation);
+        break;
+    case to_underlying(OperationalState::ErrorStateEnum::kCommandInvalidInState):
+        err.errorStateID = to_underlying(OperationalState::ErrorStateEnum::kCommandInvalidInState);
+        break;
+    default:
+        err.errorStateID = to_underlying(OperationalState::ErrorStateEnum::kUnknownEnumValue);
+        break;
+    }
+
+    GetOperationalStateInstance()->OnOperationalErrorDetected(err);
+
+    return CHIP_NO_ERROR;
+}
+
+/********************************************************
+ * RVC Operational State Functions
+ *********************************************************/
+
+CHIP_ERROR ManualRVCOperationalStateCommandHelpHandler(int argc, char ** argv)
+{
+    sShellManualRVCOperationalStateSubCommands.ForEachCommand(Shell::PrintCommandHelp, nullptr);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ManualRVCOperationalStateCommandHandler(int argc, char ** argv)
+{
+    if (argc == 0)
+    {
+        return ManualRVCOperationalStateCommandHelpHandler(argc, argv);
+    }
+
+    return sShellManualRVCOperationalStateSubCommands.ExecCommand(argc, argv);
+}
+
+CHIP_ERROR ManualRVCOperationalStateSetStateCommandHandler(int argc, char ** argv)
+{
+    if (argc != 1)
+    {
+        return ManualRVCOperationalStateCommandHelpHandler(argc, argv);
+    }
+    uint32_t state = atoi(argv[0]);
+
+    CHIP_ERROR err;
+    err = GetRVCOperationalStateInstance()->SetOperationalState(state);
+
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "ManualRVCOperationalStateSetStateCommandHandler Failed!\r\n");
+    }
+
+    return err;
+}
+
+CHIP_ERROR ManualRVCOperationalStateSetErrorCommandHandler(int argc, char ** argv)
+{
+    if (argc != 1)
+    {
+        return ManualRVCOperationalStateCommandHelpHandler(argc, argv);
+    }
+
+    GenericOperationalError err(to_underlying(ErrorStateEnum::kNoError));
+    uint32_t error = atoi(argv[0]);
+
+    switch (error)
+    {
+    case to_underlying(OperationalState::ErrorStateEnum::kNoError): // 0x00, 0
+        err.errorStateID = to_underlying(OperationalState::ErrorStateEnum::kNoError);
```

Comment:
As for ManualOperationalStateSetErrorCommandHandler, seems like this could be a lot more concise and readable...

Reply:
Updated to more readable switch case